oak_runtime/src/node/http/mod.rs

```diff
@@ -488,24 +503,26 @@ impl HttpResponseIterator {
         response_receiver.receive(&self.runtime)
     }
 
-    fn to_response(&self) -> Response<Body> {
+    fn to_response(&self) -> Result<Response<Body>, http::Error> {
         info!(
             "Generating response for runtime {} and reader {:?}.",
             self.runtime.node_id.0, self.response_reader
         );
-        let mut response = Response::new(Body::empty());
         match self.read_response() {
             Ok(http_response) => {
                 let status_code = http_response.status as u16;
-                *response.body_mut() = Body::from(http_response.body);
-                *response.status_mut() = StatusCode::from_u16(status_code)
+                let status_code = StatusCode::from_u16(status_code)
                     .unwrap_or_else(|_| panic!("Error when creating status code {}", status_code));
+                http::response::Builder::new()
+                    .status(status_code)
```

Comment:
I think since now `body` is actually doing the check on everything from the builder, you can just pass `http_response.status` here, no need to parse it first. If it fails to parse, it should result in an http::Error anyways, right? If this does not work, then it may be worth adding manual logic to make the method return Err anyways.

Reply:
I don't think I can just pass `http_response.status` the types are different.

Reply:
weird, looking at the type, to me it looks like it should work: https://docs.rs/http/0.2.1/http/response/struct.Builder.html#method.status

anyways, I'm probably missing something, maybe the error types don't line up

Reply:
`http_response.status as u16` works.